### ISSUE TITLE
`constant_subexpression_elimination` not comparing initializer values

### ISSUE THREAD

**Issue by sriduth** [microsoft/onnxruntime#5585]
**Describe the bug**
The common subexpression elimination optimization introduced in #3047 does not merge nodes that can be merged.

**Urgency**
None

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): MacOS
- ONNX Runtime installed from (source or binary): Source
- ONNX Runtime version: 1.5.2 (master)
- Python version: 3.7
- Visual Studio version (if applicable):
- GCC/Compiler version (if compiling from source): clang-11
- CUDA/cuDNN version: -
- GPU model and memory: -

**To Reproduce**
- To reproduce using a unit test, add the following test
```c++
TEST(CseTests, CSEInitializer) {
  auto model_uri = "source_model.onnx"; // this has initializers in input
  std::shared_ptr<Model> model;
  ASSERT_TRUE(Model::Load(model_uri, model, nullptr,\
                          DefaultLoggingManager().DefaultLogger()).IsOK());
  
  ApplyCse(*model);
  
  Model::Save(*model, "dst_model.onnx");
}
```

and invoke the test runner as follows: `./onnxruntime_test_all --gtest_filter="CseTests.CSEInitializer"`

- Attach the ONNX model to the issue (where applicable) to expedite investigation 

**Expected behavior**
The output of the batch norm layer is fed to multple pads, which are fed itno multiple AveragePools which can be merged

**Screenshots**

After removing initializers:

<img width="645" alt="Screenshot 2020-10-24 at 11 32 09 AM" src="https://user-images.githubusercontent.com/4116675/97069239-8ef7c600-15ec-11eb-8e84-190d46696b6e.png">

Before removing initializers:

<img width="981" alt="Screenshot 2020-10-24 at 11 30 50 AM" src="https://user-images.githubusercontent.com/4116675/97069220-5fe15480-15ec-11eb-8c8c-aaa99794bb5e.png">

**Additional context**
Add any other context about the problem here. If the issue is about a particular model, please share the model details as well to facilitate debugging.

Applying the following change would fix it - 
```c++
@@ -363,6 +372,9 @@ Status CommonSubexpressionElimination::ApplyImpl(Graph& graph, bool& modified, i
     std::vector<const EquivalenceClass*> input_values;
     input_values.reserve(node->InputDefs().size());
     for (const NodeArg* input_def : node->InputDefs()) {
+      if (graph.IsInitializedTensor(input_def->Name()))
+        continue;
+
```

**Comment by sriduth** [comment]
Is it reccomended to remove initializers as inputs when using optimisations from onnxruntime? The model was obtained by exporting from pytorch.

**Comment by sriduth** [comment]
Looks like removing initializer inputs will also merge sibling nodes that belong to the same equivalence class that have different trained parameters - will investigate

**Comment by sriduth** [comment]
https://github.com/sriduth/onnxruntime/tree/fix/cse_with_initializer adds changes to compare the initializer value corresponding to a `NodeDef*` if `graph.IsInitializedTensor()`

**Comment by sriduth** [comment]
Opened a PR which also compares the initializers when constructing and checking for equality of 2 `EquivalenceClass`es.
